feature/google/src/bc5SupportDev/java/com/revenuecat/purchases/google/BillingWrapper.kt

```diff
@@ -0,0 +1,731 @@
+        executeRequestOnUIThread { connectionError ->
+            if (connectionError == null) {
+                withConnectedClient {
+                    acknowledgePurchase(
+                        AcknowledgePurchaseParams.newBuilder().setPurchaseToken(token).build()
+                    ) { billingResult ->
+                        onAcknowledged(billingResult, token)
+                    }
+                }
+            }
+        }
+    }
+
+    override fun queryPurchases(
+        appUserID: String,
+        onSuccess: (Map<String, StoreTransaction>) -> Unit,
+        onError: (PurchasesError) -> Unit
+    ) {
+        withConnectedClient {
+            log(LogIntent.DEBUG, RestoreStrings.QUERYING_PURCHASE)
+
+            val querySubsPurchasesParams =
+                QueryPurchasesParams.newBuilder().setProductType(BillingClient.ProductType.SUBS).build()
+            this.queryPurchasesAsync(querySubsPurchasesParams) querySubPurchasesAsync@{
+                    activeSubsResult, activeSubsPurchases ->
+
+                if (!activeSubsResult.isSuccessful()) {
+                    val purchasesError = activeSubsResult.responseCode.billingResponseToPurchasesError(
+                        RestoreStrings.QUERYING_SUBS_ERROR.format(activeSubsResult.toHumanReadableDescription())
+                    )
+                    onError(purchasesError)
+                    return@querySubPurchasesAsync
+                }
+
+                val mapOfActiveSubscriptions =
+                    activeSubsPurchases.toMapOfGooglePurchaseWrapper(BillingClient.ProductType.SUBS)
+
+                val queryInAppsPurchasesParams =
+                    QueryPurchasesParams.newBuilder().setProductType(BillingClient.ProductType.INAPP).build()
+                this.queryPurchasesAsync(queryInAppsPurchasesParams) queryInAppsPurchasesAsync@{
+                        unconsumedInAppsResult, unconsumedInAppsPurchases ->
+                    if (!unconsumedInAppsResult.isSuccessful()) {
+                        val purchasesError = unconsumedInAppsResult.responseCode.billingResponseToPurchasesError(
+                            RestoreStrings.QUERYING_INAPP_ERROR.format(
+                                unconsumedInAppsResult.toHumanReadableDescription()
+                            )
+                        )
+                        onError(purchasesError)
+                        return@queryInAppsPurchasesAsync
+                    }
+                    val mapOfUnconsumedInApps =
+                        unconsumedInAppsPurchases.toMapOfGooglePurchaseWrapper(BillingClient.ProductType.INAPP)
+                    onSuccess(mapOfActiveSubscriptions + mapOfUnconsumedInApps)
+                }
+            }
+        }
+    }
+
+    private fun List<Purchase>.toMapOfGooglePurchaseWrapper(
+        @BillingClient.ProductType productType: String
+    ): Map<String, StoreTransaction> {
+        return this.associate { purchase ->
```

Comment:
I didn't know about this `associate`

Reply:
androidstudio suggested it to me!

Reply:
Should we apply this as well to the `latestDependencies` version?